syntax = "proto3";

package tradeoperation;

// Service to check server health and connectivity
service ConnectivityService {
  rpc CheckServerHealth (NetworkInfoRequest) returns (NetworkInfoResponse);
}

message NetworkInfoRequest {
  string client_ip = 1; // Client IP to check server connectivity
}

message NetworkInfoResponse {
  string message = 1;  // Message indicating the status
  bool is_healthy = 2; // True if server is healthy
}


// Service to execute commands over Telnet
service TelnetService {
    rpc ExecuteCommand (CommandRequest) returns (CommandResponse);
}

message CommandRequest {
    string command = 1; // Command to execute over Telnet
}

message CommandResponse {
    string result = 1; // The result of the command execution
}

